### ISSUE TITLE
nvim stuck in insert mode

### ISSUE THREAD

**Issue by 1ak31sha** [neovim/neovim#14759]
- `nvim --version`: 0.4.4
- Operating system/version: macOS catalina
- Terminal name/version: iterm2
- `$TERM`: fish: Unknown command: screen-256color

[ ] `vim -u DEFAULTS` (version: ) behaves differently
- yes, i can paste the text no problems in vim

### Steps to reproduce using `nvim -u NORC`

- nvim -u NORC -> yes it also happens with no rc

paste the following text:

```
- [ n/a] Tests for the changes have been added (for bug fixes / features)
- [ n/a] Documentation has been added where required
- [ n/a] Story coverage (Storybook)
- [ n/a] Accessibility support has been tested
- [n/a] Responsive layout and functionality have been tested
  - [n/a] desktop and mobile view responsiveness are supported
  - [n/a] no loss of functionality or content when zoomed in to 400%
  - [n/a] no horizontal scrolling when zoomed in to 400% (only vertical scrolling, exception: DataGrids)
- [n/a] Cross-browser support has been tested(Chrome, Firefox, Safari, Edge)
- [n/a] i18n support (components only) has been added(uses `TranslationProvider`)
- [n/a] avoids using many levels of prop drilling
- [n/a] check console on errors and warnings
```

i would now be stuck in insert mode, cant figure out any way to get out, and this is what the buffers got in it:
```
- [ n/a] Tests for the changes have been added (for bug fixes / features)
- [ n/a] Documentation has been added where required
- [ n/a] Story coverage (Storybook)
- [ n/a] Accessibility support has been tested
- [n/a] Responsive layout and functionality have been tested
  - [n/a] desktop and mobile view responsiveness are supported
  - [n/a] no loss of functionality or content when zoomed in to 400%
  - [n/a] no horizontal scrolling when zoomed in to 400% (only vertical scrolling, exception: DataGrids)
- [n/a] Cross-browser support has been tested(Chrome, Firefox, Safari, Edge)
- [n/a] i18n support (components only) has been added(uses `TranslationProvider`)
- [n/a] avoids using many levels of prop drilling
- [n/a] check console on errors and warnings^[[201~
```

```

### Actual behaviour
stuck in insert mode

### Expected behaviour
not stuck in insert mode with weird `^[[201~`

**Comment by clason** [comment]
Can you reproduce this with one of the nightlies (or  `brew install neovim --HEAD`)? I'm on macOS with iTerm2 and fish and can't reproduce it.

(Also, `NORC` still sources your plugins; you need `NONE` instead or `--noplugin` in addition.)

**Comment by 1ak31sha** [comment]
> Can you reproduce this with one of the nightlies (or `brew install neovim --HEAD`)? I'm on macOS with iTerm2 and fish and can't reproduce it.
> 
> (Also, `NORC` still sources your plugins; you need `NONE` instead or `--noplugin` in addition.)

yea i actualy did  it with `nvim -u NONE` and it still happened. will give the nightly a try now

**Comment by 1ak31sha** [comment]
the error didnt happen on:
```
❯ nvim -v
NVIM v0.5.0-dev+1390-g988f88c24
Build type: Release
LuaJIT 2.1.0-beta3
Compilation: clang -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=1 -O2 -DNDEBUG -Wall -Wextra -pedantic -Wno-unused-parameter -Wstrict-prototypes -std=gnu99 -Wshadow -Wconversion -Wmissing-prototypes -Wimplicit-fallthrough -Wvla -fstack-protector-strong -fno-common -fdiagnostics-color=auto -DINCLUDE_GENERATED_DECLARATIONS -D_GNU_SOURCE -DNVIM_MSGPACK_HAS_FLOAT32 -DNVIM_UNIBI_HAS_VAR_FROM -DMIN_LOG_LEVEL=3 -I/tmp/neovim-20210610-70315-fvmcc6/build/config -I/tmp/neovim-20210610-70315-fvmcc6/src -I/usr/local/include -I/Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk/usr/include -I/usr/local/opt/gettext/include -I/tmp/neovim-20210610-70315-fvmcc6/build/src/nvim/auto -I/tmp/neovim-20210610-70315-fvmcc6/build/include
Compiled by lakeishapogrzeba@TR-C02VD7A3HTD6

```
@clason

**Comment by 1ak31sha** [comment]
@clason  perhaps youd want to update the PR template to ask reporters to use `nvim -u NONE` instead of NORC.

**Comment by clason** [comment]
Yeah, I can't reproduce with that version, either. It would help if you could try to make these steps more minimal. (Try with a clean environment -- rename `.config/nvim` and `.local/share/nvim` and try to find the **minimal** text that triggers it. The exact keystrokes you use to enter insert mode help as well.)

**Comment by clason** [comment]
Also check the version (and settings) for iTerm2; maybe the issue is with _that_ hanging.

**Comment by 1ak31sha** [comment]
@clason how much more minimal can the steps be? you launch vim with `-u NONE` and paste the text from the description. i dont really have time atm to try every exact combination of text, sorry
and, perhaps its simply fixed in neovim 5.0-HEAD?

**Comment by clason** [comment]
Oh, sorry, I completely overlooked that you wrote "did**nt**" happen. Yes, then it's clearly fixed, and not worth spending more time on -- we hope to put out 0.5 (with that fix) Real Soon(tm).
